pkg/platform/kube/platform.go

```diff
@@ -1199,3 +1213,53 @@ func (p *Platform) enrichTriggerWithServiceType(functionConfig *functionconfig.C
 
 	return trigger
 }
+
+func (p *Platform) validateAPIGatewayFunctionsHaveNoIngresses(platformAPIGateway *platform.APIGatewayConfig) error {
+
+	// check ingresses on every upstream function
+	errGroup, _ := errgroup.WithContext(context.TODO())
+	for _, upstream := range platformAPIGateway.Spec.Upstreams {
```

Comment:
It would fail because https://golang.org/doc/faq#closures_and_goroutines

add a test to catch such scenario (TDD it) and fix it ^